File: tournaments/swapDiagonals/swapDiagonals.js
function swapDiagonals(matrix) {
    const n = matrix.length - 1;

    for (let i = 0; (n + 1) / 2 > i; ++i) {
        [matrix[i][i], matrix[i][n - i]] = [matrix[i][n - i], matrix[i][i]];
        [matrix[n - i][i], matrix[n - i][n - i]] = [matrix[n - i][n - i], matrix[n - i][i]];
    }

    return matrix;
}

File: tournaments/swapDiagonals/swapDiagonals.cpp
std::vector<std::vector<int>> swapDiagonals(std::vector<std::vector<int>> matrix) {
    for (int i = 0; i < matrix.size(); ++i)
		std::swap(matrix[i][matrix.size() - i - 1], matrix[i][i]);
	return matrix;
}
